frontend/database.go

```diff
@@ -153,3 +155,58 @@ func (d *DBClient) DeleteClusterDoc(ctx context.Context, resourceID string, part
 	}
 	return nil
 }
+
+// GetSubscriptionDoc retreives a subscription document from async DB using the subscription ID
+func (d *DBClient) GetSubscriptionDoc(ctx context.Context, partitionKey string) (*SubscriptionDocument, bool, error) {
+	container, err := d.client.NewContainer(d.config.DBName, subsContainer)
+	if err != nil {
+		return nil, false, err
+	}
+
+	query := "SELECT * FROM c WHERE c.partitionKey = @partitionKey"
+	opt := azcosmos.QueryOptions{
+		PageSizeHint:    1,
+		QueryParameters: []azcosmos.QueryParameter{{Name: "@partitionKey", Value: partitionKey}},
+	}
+
+	pk := azcosmos.NewPartitionKeyString(partitionKey)
+	queryPager := container.NewQueryItemsPager(query, pk, &opt)
+
+	var doc *SubscriptionDocument
+	for queryPager.More() {
+		queryResponse, err := queryPager.NextPage(ctx)
+		if err != nil {
+			return nil, false, err
+		}
+
+		for _, item := range queryResponse.Items {
+			err = json.Unmarshal(item, &doc)
+			if err != nil {
+				return nil, false, err
+			}
+		}
+	}
+	if doc != nil {
+		return doc, true, nil
+	}
+	return nil, false, nil
```

Comment:
Thinking out loud, for this and `GetClusterDoc` should "not found" always return an error? Do we need to boolean return to identify if an object was found or not?

Reply:
Oh, yeah, we could do https://go.dev/blog/go1.13-errors instead of string comparison if we wanted to get rid of the boolean return value.

Reply:
Sounds like a plan, ill work on getting this updated to define an error and check the error type instead.

Reply:
@mjlshen updated, lemme know what you think, its better IMO. Cleaner.

Reply:
+1 on eliminating the boolean, but as I noted elsewhere be careful in error handling to check specifically for `NotFoundError` _before_ `err != nil`.